### PR TITLE
Add Elasticsearch index cleaner as cron job

### PR DESCRIPTION
**Pull request by pavolloffay**
CronJob for https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/esCleaner.py#L11

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/155)

File: test/elasticsearch.yml [hunk 1]
```diff
@@ -16,27 +16,21 @@ spec:
     spec:
       containers:
         - name: elasticsearch
-          image: docker.elastic.co/elasticsearch/elasticsearch:5.6.0
+          image: docker.elastic.co/elasticsearch/elasticsearch:5.6.12
```
Comment (pavolloffay) [inline test/elasticsearch.yml]: I have bumped patch version. And disabled xpack, we should not use it anyways. There an official ES6x without licensed xpack, but I would rather keep using the lowest supported version.

File: pkg/apis/io/v1alpha1/jaeger_types.go [hunk 2]
```diff
@@ -150,6 +151,14 @@ type JaegerDependenciesSpec struct {
 	ElasticsearchNodesWanOnly   bool   `json:"elasticsearchNodesWanOnly"`
 }
 
+// JaegerEsIndexCleanerSpec holds the options related to es-index-cleaner
+type JaegerEsIndexCleanerSpec struct {
+	Enabled      bool   `json:"enabled"`
+	NumberOfDays int    `json:"numberOfDays"`
```
Comment (pavolloffay) [inline pkg/apis/io/v1alpha1/jaeger_types.go]: Used 7 days as default. TTL in c* is 2 day though...

File: pkg/apis/io/v1alpha1/jaeger_types.go [hunk 3]
```diff
@@ -150,6 +151,14 @@ type JaegerDependenciesSpec struct {
 	ElasticsearchNodesWanOnly   bool   `json:"elasticsearchNodesWanOnly"`
 }
 
+// JaegerEsIndexCleanerSpec holds the options related to es-index-cleaner
+type JaegerEsIndexCleanerSpec struct {
+	Enabled      bool   `json:"enabled"`
```
Comment (pavolloffay) [inline pkg/apis/io/v1alpha1/jaeger_types.go]: By default disabled. 

I think we should enable it by default along with dependencies job. I would rather do it in a separate PR to keep this minimal.

File: pkg/apis/io/v1alpha1/jaeger_types.go (hunk 2 above)
Comment (pavolloffay) [reply pkg/apis/io/v1alpha1/jaeger_types.go]: The script supports other configuration properties, see https://github.com/jaegertracing/jaeger/blob/master/plugin/storage/es/esCleaner.py#L11

index prefix, es hostname, I have not exposed them directly but they are derived from storage properties map.
